pkg/register/component.go

```diff
@@ -0,0 +1,227 @@
+		return "", err
+	}
+
+	// Generate an unused
+	var username string
+	username, err = c.generateUnusedUsername(ctx, realmName)
+	if err != nil {
+		return "", err
+	}
+
+	// Create the user in Keycloak
+	var kcUser = user.UserToKcUserRepresentation()
+	kcUser.Username = &username
+	kcUser.RequiredActions = realmConf.RegisterRequiredActions
+	var userID string
+	userID, err = c.keycloakClient.CreateUser(accessToken, AdminRealm, realmName, kcUser)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Failed to create user through Keycloak API", "err", err.Error())
+		return "", err
+	}
+
+	// Create a DB transaction
+	var tx database.Transaction
+	tx, err = c.usersDBModule.BeginTransaction(ctx)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't begin transaction", "err", err.Error())
+		return "", err
+	}
+	defer tx.Close()
+
+	// Store user in database
+	err = c.usersDBModule.StoreOrUpdateUser(ctx, realmName /* realmID */, apiregister.DBUser{
+		UserID:               &userID,
+		Gender:               user.Gender,
+		BirthDate:            user.BirthDate,
+		BirthLocation:        user.BirthLocation,
+		IDDocumentType:       user.IDDocumentType,
+		IDDocumentNumber:     user.IDDocumentNumber,
+		IDDocumentExpiration: user.IDDocumentExpiration,
+	})
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't store user details in database", "err", err.Error())
+		return "", err
+	}
+
+	// Send execute actions email
+	if realmConf.RegisterRequiredActions != nil && len(*realmConf.RegisterRequiredActions) > 0 {
+		err = c.keycloakClient.ExecuteActionsEmail(accessToken, "master", userID, *realmConf.RegisterRequiredActions)
+		if err != nil {
+			c.logger.Warn(ctx, "msg", "ExecuteActionsEmail failed", "err", err.Error())
+			return "", err
+		}
+	}
+
+	// Commit the DB transaction
+	err = tx.Commit()
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't commit transaction", "err", err.Error())
+		return "", errorhandler.CreateInternalServerError("database")
+	}
+
+	// store the API call into the DB
+	c.reportEvent(ctx, "REGISTER_USER", database.CtEventRealmName, realmName, database.CtEventUserID, userID)
+
+	return username, nil
+}
+
+func (c *component) checkExistingUser(ctx context.Context, accessToken, realmName string, user apiregister.User) error {
+	// Email already used?
+	var existingUser, err = c.usersDBModule.GetUser(ctx, realmName, nil, nil, user.EmailAddress)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't get user from db", "err", err.Error())
+		return errorhandler.CreateInternalServerError("database")
+	}
+	if existingUser == nil {
+		// Free to go on processing this user creation
+		return nil
+	}
+
+	var kcUser kc.UserRepresentation
+	kcUser, err = c.keycloakClient.GetUser(accessToken, realmName, *existingUser.UserID)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't get user from keycloak", "err", err.Error())
+		return errorhandler.CreateInternalServerError("keycloak")
+	}
+
+	if kcUser.Enabled == nil || *kcUser.Enabled {
```

Comment:
Should we rely on Enabled property ? 
If the user is disabled, I am not sure he will be able to perform email actions.
Maybe we should consider to rely on emailVerified or a specific attribute